t/tnt/init.lua

```diff
@@ -37,6 +53,116 @@ local function tnt_prepare(cfg_args)
     box.cfg(cfg_args)
 end
 
+-- Creates master and replica setup for queue states switching tests.
+local function tnt_cluster_prepare(cfg_args)
+    -- Prepare master.
+    cfg_args = cfg_args or {}
+    local files = fio.glob(fio.pathjoin(dir, '*'))
+    for _, file in pairs(files) do
+        if fio.basename(file) ~= 'tarantool.log' then
+            log.info("skip removing %s", file)
+            fio.unlink(file)
+        end
+    end
+
+    cfg_args['wal_dir']         = dir
+    cfg_args['read_only']       = false
+    cfg_args[snapdir_optname()] = dir
+    cfg_args[logger_optname()]  = fio.pathjoin(dir, 'tarantool.log')
+    cfg_args['listen']          = bind_master
+    cfg_args['replication']     = {'replicator:password@' .. bind_replica,
+                                   'replicator:password@' .. bind_master}
+    if vinyl_name() then
+        local vinyl_optname     = vinyl_name() .. '_dir'
+        cfg_args[vinyl_optname] = dir
+    end
+    cfg_args['replication_connect_quorum']  = 1
+    cfg_args['replication_connect_timeout'] = 0.01
+
+    box.cfg(cfg_args)
+    -- Allow guest all operations.
+    box.schema.user.grant('guest', 'read, write, execute', 'universe')
+    box.schema.user.create('replicator', {password = 'password'})
+    box.schema.user.grant('replicator', 'replication')
+
+    -- Prepare replica.
+    dir_replica = fio.tempdir()
+
+    local vinyl_opt = nil
+    if vinyl_name() then
+        vinyl_opt = ', ' .. vinyl_name() .. '_dir = \'' .. dir_replica .. '\''
+    else
+        vinyl_opt = ''
+    end
+
+    local cmd_replica = {
+        arg[-1],
+        '-e',
+        [[
+        box.cfg {
+            read_only = true,
+            replication = 'replicator:password@]] .. bind_master ..
+            '\', listen = \'' .. bind_replica ..
+            '\', wal_dir = \'' .. dir_replica ..
+            '\', ' .. snapdir_optname() .. ' = \'' .. dir_replica ..
+            '\', ' .. logger_optname() .. ' = \'' ..
+            fio.pathjoin(dir_replica, 'tarantool.log') .. '\'' ..
+            vinyl_opt ..
+        '}'
+    }
+
+    replica = popen.new(cmd_replica, {
+        stdin = 'devnull',
+        stdout = 'devnull',
+        stderr = 'devnull',
+    })
```

Comment:
Beware: tarantool-1.10 has no `popen` module.

```shell
$ (needle='popen: add popen Lua module'; for branch in 1.7 1.9 1.10 2.1 2.2 2.3 2.4 2.5 2.6 2.7 2.8 2.10 master; do printf "${branch}\t"; commit_description=$(git log --oneline "origin/${branch}" | grep -F "${needle}"); [ -n "${commit_description}" ] && echo "$(git describe --long --always $(echo "${commit_description}" | cut -d" " -f1)) $(echo "${commit_description}" | cut -d" " -f2-)" || echo "(not found)"; done)

1.7	(not found)
1.9	(not found)
1.10	(not found)
2.1	(not found)
2.2	(not found)
2.3	(not found)
2.4	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
2.5	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
2.6	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
2.7	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
2.8	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
2.10	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
master	2.4.0-237-g6d1c5ff52 popen: add popen Lua module
```

You can either use `os.execute()` or skip the test on tarantool versions without `popen`. Any way is okay for me.

Reply:
Missed the relevant commit, sorry. Now I see that it is done.